### ISSUE TITLE
[mobile] config.xml is not generated correctly right after android platform is added

### ISSUE THREAD

**Issue by mrsharkbait** [meteor/meteor#8338]
This issue is found in Meteor 1.4.2.3 running on Mac OS X Sierra 10.12.2

**Issue:** If `mobile-build.js` exists, and then the Android platform is added to the project, the generated `config.xml` in turn does not generate the web application bundle (not sure if this is the correct terminology).  After a `meteor build`, the bundle `android/project/assets/www/application` does not get created.

In Meteor 1.3, this bundle (the `application` folder) is created properly.

**Use case:** I have a project where multiple APKs are created, each with a different app ID.  The workflow is as follows:
The file mobile-config.js is modified with App info.

1. Run `meteor remove-platform android` - To completely clear previous App info
2. Run `meteor reset` - Necessary to remove old `config.xml`
3. Run `meteor add-platform android` - Create new `config.xml`
4. Run `meteor build` - Used to create APKs

**To reproduce:**

1. Create empty Meteor project: `meteor create --bare <somewhere>`
2. Create `mobile-build.js`.  Here is a simple one:
```
App.info({
id: 'com.foo',
name: 'BareMobile',
description: 'Customizing the bare project',
author: 'MrSharkbait',
email: 'eugene@foo.com',
website: 'http://www.foo.com',
version: '1.0.0'
});
```
3. Add Android platform: `meteor add-platform android` Note the contents of `config.xml`:
```
<?xml version='1.0' encoding='utf-8'?>
<widget id="com.foo" version="1.0.0" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">
    <name>BareMobile</name>
    <description>Customizing the bare project</description>
    <author email="eugene@foo.com" href="http://www.foo.com">MrSharkbait</author>
    <preference name="webviewbounce" value="false" />
    <preference name="DisallowOverscroll" value="true" />
    <content src="http://localhost:12128/" />
    <allow-intent href="tel:*" />
    <allow-intent href="geo:*" />
    <allow-intent href="mailto:*" />
    <allow-intent href="sms:*" />
    <allow-intent href="market:*" />
    <allow-intent href="itms:*" />
    <allow-intent href="itms-apps:*" />
    <allow-navigation href="http://localhost" />
    <platform name="ios" />
    <platform name="android" />
</widget>
```
4. Do a build `meteor build ../bareBuild-foo --server=https://app.foo.com` Note that `../bareBuild-foo/android/project/assets/www/application` is missing.  The resulting build in `../bareBuild-foo` is unusable.
After build is complete, look at `config.xml`:
```
<?xml version="1.0"?>
<widget id="com.bar" version="1.0.0" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">
  <name>ScaffoldMobile</name>
  <description>Customizing the scaffold</description>
  <author href="http://www.techshaman.com" email="eugene@techshaman.com">MrSharkbait</author>
  <preference name="webviewbounce" value="false"/>
  <preference name="DisallowOverscroll" value="true"/>
  <preference name="AutoHideSplashScreen" value="false"/>
  <preference name="SplashScreen" value="screen"/>
  <preference name="SplashScreenDelay" value="5000"/>
  <preference name="FadeSplashScreenDuration" value="250"/>
  <preference name="ShowSplashScreenSpinner" value="false"/>
  <preference name="StatusBarOverlaysWebView" value="false"/>
  <preference name="StatusBarStyle" value="default"/>
  <content src="http://localhost:12128/"/>
  <allow-intent href="tel:*"/>
  <allow-intent href="geo:*"/>
  <allow-intent href="mailto:*"/>
  <allow-intent href="sms:*"/>
  <allow-intent href="market:*"/>
  <allow-intent href="itms:*"/>
  <allow-intent href="itms-apps:*"/>
  <allow-navigation href="http://localhost"/>
  <access origin="*://app.foo.com"/>
  <platform name="ios">
    <icon src="resources/iphone_2x.icon.png" width="120" height="120"/>
    <icon src="resources/iphone_3x.icon.png" width="180" height="180"/>
    <icon src="resources/ipad.icon.png" width="76" height="76"/>
    <icon src="resources/ipad_2x.icon.png" width="152" height="152"/>
    <icon src="resources/ipad_pro.icon.png" width="167" height="167"/>
    <icon src="resources/ios_settings.icon.png" width="29" height="29"/>
    <icon src="resources/ios_settings_2x.icon.png" width="58" height="58"/>
    <icon src="resources/ios_spotlight.icon.png" width="40" height="40"/>
    <icon src="resources/ios_spotlight_2x.icon.png" width="80" height="80"/>
    <splash src="resources/iphone_2x.splash.png" width="640" height="960"/>
    <splash src="resources/iphone5.splash.png" width="640" height="1136"/>
    <splash src="resources/iphone6.splash.png" width="750" height="1334"/>
    <splash src="resources/iphone6p_portrait.splash.png" width="1242" height="2208"/>
    <splash src="resources/iphone6p_landscape.splash.png" width="2208" height="1242"/>
    <splash src="resources/ipad_portrait.splash.png" width="768" height="1024"/>
    <splash src="resources/ipad_portrait_2x.splash.png" width="1536" height="2048"/>
    <splash src="resources/ipad_landscape.splash.png" width="1024" height="768"/>
    <splash src="resources/ipad_landscape_2x.splash.png" width="2048" height="1536"/>
  </platform>
  <platform name="android">
    <icon src="resources/android_mdpi.icon.png" width="48" height="48"/>
    <icon src="resources/android_hdpi.icon.png" width="72" height="72"/>
    <icon src="resources/android_xhdpi.icon.png" width="96" height="96"/>
    <icon src="resources/android_xxhdpi.icon.png" width="144" height="144"/>
    <icon src="resources/android_xxxhdpi.icon.png" width="192" height="192"/>
    <splash src="resources/android_mdpi_portrait.splash.png" width="320" height="470" density="port-mdpi"/>
    <splash src="resources/android_mdpi_landscape.splash.png" width="470" height="320" density="land-mdpi"/>
    <splash src="resources/android_hdpi_portrait.splash.png" width="480" height="640" density="port-hdpi"/>
    <splash src="resources/android_hdpi_landscape.splash.png" width="640" height="480" density="land-hdpi"/>
    <splash src="resources/android_xhdpi_portrait.splash.png" width="720" height="960" density="port-xhdpi"/>
    <splash src="resources/android_xhdpi_landscape.splash.png" width="960" height="720" density="land-xhdpi"/>
    <splash src="resources/android_xxhdpi_portrait.splash.png" width="1080" height="1440"/>
    <splash src="resources/android_xxhdpi_landscape.splash.png" width="1440" height="1080"/>
  </platform>
</widget>
```
5. To repeat, run 
```
meteor remove-platform android
meteor reset
```
Go to Step 3

**Workaround:**
Running `meteor build` one more time will result in `../bareBuild-foo/android/project/assets/www/application` folder created correctly.

**Comment by abernix** [comment]
Thank you for documenting this so thoroughly!  This should actually be fixed in Meteor 1.4.2.6 (just released) thanks to https://github.com/meteor/meteor/pull/8142 (see related issues in there, too).  Do you mind trying it and reporting back?

    meteor update --release 1.4.2.6

**Comment by mrsharkbait** [comment]
Aigh!  That issue eluded my search. I will test out 1.4.2.6 and report back.

e
